src/sentry/profiles/profile_chunks.py

```diff
@@ -57,21 +57,29 @@ def get_chunk_ids(
     return [row["chunk_id"] for row in result["data"]]
 
 
-def resolve_datetime64(value: datetime | None, precision: int = 6) -> Function | None:
+def resolve_datetime64(
+    raw_value: datetime | str | float | None, precision: int = 6
+) -> Function | None:
```

Comment:
For a follow up, but let's move this somewhere that can be used by more than just profiles, Spans will be using a `datetime64(6)` soon too

Reply:
Will follow up on this in another PR. Ideally this should live in the SDK but this will serve as a stop-gap solution.